### ISSUE TITLE
net/http or fasthttp?

### ISSUE THREAD

**Issue by kiwi2019** [kataras/iris#1288]
Is iris based on standard net/http or fasthttp? I see iris.go

```
func Addr(addr string, hostConfigs ...host.Configurator) Runner {
	return func(app *Application) error {
		return app.NewHost(&http.Server{Addr: addr}).
			Configure(hostConfigs...).
			ListenAndServe()
	}
}
```
it use http.Server is standard net/http！

How can I use the fasthttp ？

thanks...

**Comment by kataras** [comment]
Iris is based on net/http for more than two years now, if you want fasthttp take a look at the `5.0.0` branch but it is not supported any more, however, there is code inside that helps you understand and implement things like session manager, websockets, view managment and etc.

**Comment by kiwi2019** [comment]
> Iris is based on net/http for more than two years now, if you want fasthttp take a look at the `5.0.0` branch but it is not supported any more, however, there is code inside that helps you understand and implement things like session manager, websockets, view managment and etc.

Isn't fasthttp performance better?
Why iris use net/http?
What is the consideration?
Finally, thank you very much...I use iris for my project ,it very easy to use and strong

**Comment by kataras** [comment]
I did explain when we moved from fasthttp to net/http at go 1.7. But in short: net/http supports http/2 and it does it very well, fasthttp does not. But **the most important reason is that fasthttp is not maintained as it was anymore**, the author of that library doesn't work on this project for a long time now, more than a whole year, valyala was really an experienced developer. Hope I answered your question.

**Comment by kataras** [comment]
I did explain when we moved from fasthttp to net/http at go 1.7. But in short: net/http supports http/2 and it does it very well, fasthttp does not. Secondly, many 3rd party libraries are written for net/http so you are able to use them with Iris (look the [_examples](https://github.com/kataras/iris/tree/master/_examples) repository's directory). However, **the most important reason is that fasthttp is not maintained as it was anymore**, the author of that library doesn't work on this project for a long time now, more than a whole year, valyala was really an experienced developer. Hope I answered your question.
